pkg/common/fault/constants.go

```diff
@@ -34,6 +34,13 @@ const (
 	// CSIApiServerOperationFault is the fault type when Get(), List() and others fail on the API Server
 	CSIApiServerOperationFault = "csi.fault.ApiServerOperation"
 
+	// CSIPVNotFoundFault is the fault type when PV name is not found in PVC Spec.
+	// This can happen at the time of guest cluster creation when user specifies volumes to be created
+	// in the guest cluster spec. Volume creation in such cases are typically initiated by
+	// vmoperator and an error is observed because the volume name is updated in the VM spec, even
+	// before the volume is provisioned in supervisor cluster.
+	CSIPVNotFoundFault = "csi.fault.nonstorage.PVNotFound"
```

Comment:
nit: Does it make sense to include the suspecting component in fault string as well?
For instance: `csi.fault.nonstorage.PVNotFound` -> `csi.fault.nonstorage.vmOperator.PVNotFound`

If we decide to do this, we might have to make similar change for other nonstorage faults.

Reply:
Although we know the initiator of this error workflow, the component which emits the actual fault is still CSI. So, having a prefix "csi" is indicating the component IMO. But I changed the faultString to `csi.fault.PvNotFoundInPvcSpec` as it is related to Storage
Let me know if that's reasonable.

Reply:
After discussing with Ankit, we agreed to keep the string as `csi.fault.nonstorage.PvNotFoundInPvcSpec`